crates/core-subsystem/core-resolver/src/validation/arguments_validator.rs

```diff
@@ -439,6 +450,17 @@ fn get_schema_directives(schema: &Schema, argument_name: &str) -> Vec<ConstDirec
         .collect()
 }
 
+fn is_operation_related_to_type_definition(
+    operation_name: &str,
+    schema_type_definition_name: &str,
+) -> bool {
+    let related_operations = [
+        format!("create{schema_type_definition_name}"),
```

Comment:
it seems that currently there is no other way to match operation name with schema name

Reply:
An alternative would be to set `InputValueDefinition`s directives to the validation directives. Something along the following lines:
- Change `type_validation: None,` in create_mutation_builder.rs and update_mutation_field.rs to the same as the one you have in `type_builder.rs` ``create_persistent_field`  (you can even clone the one you set in there)
- Then change `impl TypeDefinitionProvider<PostgresSubsystem> for MutationType` to put the directives. 

This is more involved, however.